src/dev/build/lib/version_info.ts

```diff
@@ -30,13 +30,23 @@ export async function getVersionInfo({ isRelease, versionQualifier, pkg }: Optio
     isRelease ? '' : '-SNAPSHOT'
   );
 
-  const buildSha = fs.existsSync(join(REPO_ROOT, '.git'))
+  const gitExists = fs.existsSync(join(REPO_ROOT, '.git'));
+
+  const buildSha = gitExists
     ? (await execa('git', ['rev-parse', 'HEAD'], { cwd: REPO_ROOT })).stdout
     : process.env.GIT_COMMIT || process.env.BUILDKITE_COMMIT || '';
 
+  // Use the date of HEAD commit if available
+  const buildDate = gitExists
```

Comment:
Thoughts on making this a build time date?  There's technically a few edge cases  where we'll swap out dependencies (e.g. pulling in a new version of metricbeat for our cloud distribution) that could be based on the same commit hash.